pkg/storage/snapshot/restore.go

```diff
@@ -1268,14 +1433,29 @@ func setLastRestoreAnnotation(restore *snapshotv1.VirtualMachineRestore, obj met
 	obj.GetAnnotations()[lastRestoreAnnotation] = getRestoreAnnotationValue(restore)
 }
 
+func getFilteredLabels(labels map[string]string) map[string]string {
+	excludedKey := backendstorage.PVCPrefix
+	excludedMap := map[string]struct{}{
+		excludedKey: {},
+	}
+
+	filteredLabels := make(map[string]string)
+	for key, value := range labels {
+		if _, excluded := excludedMap[key]; !excluded {
+			filteredLabels[key] = value
+		}
+	}
+
+	return filteredLabels
+}
+
```

Comment:
As a follow up, this can be simplified
```suggestion
func getFilteredLabels(labels map[string]string) map[string]string {
	var filteredLabels map[string]string
	for key, value := range labels {
	        if key == backendstorage.PVCPrefix {
	            continue
	        }
		filteredLabels[key] = value
	}

	return filteredLabels
}

```